sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuFileFormatDataWriter.scala

```diff
@@ -623,17 +635,23 @@ class GpuDynamicPartitionDataSingleWriter(
             newWriter(partPath, None, currentWriterStatus.fileCounter)
           currentWriterStatus.recordsInFile = 0
         }
-        val bc = GpuColumnVector.from(b.getTable(), outDataTypes)
-        closeOnExcept(bc) { _ =>
-          statsTrackers.foreach(_.newBatch(currentWriterStatus.outputWriter.path(), bc))
-          currentWriterStatus.recordsInFile += b.getRowCount()
+        splits(partIx) = null
+        withResource(part) { _ =>
+          writeTableUsingCurrentWriter(part.getTable, outDataTypes)
         }
```

Comment:
This holds onto the table too long.  The writer internally will release resources as part of writing, but this holds onto the part table throughout the entire write.  It needs to be closed as soon as we get the columnar batch out that will be handed to the writer to close.

Reply:
Will fix

Reply:
Handled this and the one below in https://github.com/NVIDIA/spark-rapids/pull/8385/commits/e9765621c62e7aa2cd03ead6a47d258898be0fb0 and https://github.com/NVIDIA/spark-rapids/pull/8385/commits/48aec9468366c44386a45908f1090d14464141f6